dsp/modules/sentence_vectorizer.py

```diff
@@ -247,4 +247,93 @@ def __call__(self, inp_examples: List["Example"]) -> np.ndarray:
         texts_to_vectorize = self._extract_text_from_examples(inp_examples)
         embeddings = self._model.embed(texts_to_vectorize, batch_size=self._batch_size, parallel=self._parallel)
 
-        return np.array([embedding.tolist() for embedding in embeddings], dtype=np.float32)
\ No newline at end of file
+        return np.array([embedding.tolist() for embedding in embeddings], dtype=np.float32)
+
+class AmazonBedrockVectorizer(BaseSentenceVectorizer):
+    '''
+    This vectorizer uses Amazon Bedrock API to convert texts to embeddings.
+    '''
+    SUPPORTED_MODELS = [
+        "amazon.titan-embed-text-v1", "amazon.titan-embed-text-v2:0",
+        "cohere.embed-english-v3", "cohere.embed-multilingual-v3"
+    ]
+
+    def __init__(
+        self,
+        model_id: str = 'amazon.titan-embed-text-v2:0',
+        embed_batch_size: int = 128,
+        region_name: str = 'us-west-2',
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+    ):
+        self.model_id = model_id
+        self.embed_batch_size = embed_batch_size
+
+        # Initialize the Bedrock client
+        self.bedrock_client = boto3.client(
```

Comment:
could this potentially be replaced with the existing [DSPy AWS model integrations](https://github.com/stanfordnlp/dspy/blob/bb97c72da6312600b8ac1a5228f79b07afc427b1/dsp/modules/aws_models.py#L24)? would be neat to tie it in!

Reply:
I'm not exactly sure what you would like to see here to be honest 🤔